src/components/ObservationForm/ObservationForm.tsx

```diff
@@ -0,0 +1,205 @@
+                )?.display,
+              },
+            ],
+          },
+          effectiveDateTime: data.effectiveDateTime.toISOString(),
+          valueQuantity: {
+            value: data.value,
+            unit: data.unit,
+          },
+        }
+        await onSubmit(obs)
+        return
+      } catch (error) {
+        console.error('Form submission error:', error)
+        throw error
+      }
+    },
+  )
+
+  const [formType, formUnit] = form.watch(['type', 'unit'])
+  const units = getObservationTypeUnits(formType)
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <Field
+        control={form.control}
+        name="status"
+        label="Status"
+        render={({ field }) => (
+          <Select onValueChange={field.onChange} {...field}>
+            <SelectTrigger>
+              <SelectValue placeholder="Status" />
+            </SelectTrigger>
+            <SelectContent>
+              {OBSERVATION_STATUS_OPTIONS.map((status) => (
+                <SelectItem key={status.code} value={status.code}>
+                  {status.display}
+                </SelectItem>
+              ))}
+            </SelectContent>
+          </Select>
+        )}
+      />
+      <Field
+        control={form.control}
+        name="type"
+        label="Type"
+        render={({ field }) => (
+          <Select
+            onValueChange={(type) => {
+              field.onChange(type)
+              form.setValue(
+                'unit',
+                getUnitOfObservationType(
+                  type as UserObservationCollection,
+                  formUnit,
+                ).unit,
+              )
+            }}
+            {...field}
+          >
+            <SelectTrigger>
+              <SelectValue placeholder="Type" />
+            </SelectTrigger>
+            <SelectContent>
+              {OBSERVATION_TYPE_OPTIONS.map((type) => (
+                <SelectItem key={type.code} value={type.code}>
+                  {type.display}
+                </SelectItem>
+              ))}
+            </SelectContent>
+          </Select>
+        )}
+      />
+      <Field
+        control={form.control}
+        name="unit"
+        label="Unit"
+        render={({ field }) => (
+          <Select {...field} key={formType}>
+            <SelectTrigger>
+              <SelectValue placeholder="Unit" />
+            </SelectTrigger>
+            <SelectContent>
+              {units?.map((unit) => (
+                <SelectItem key={unit.unit} value={unit.unit}>
+                  {unit.unit}
+                </SelectItem>
+              ))}
+            </SelectContent>
+          </Select>
+        )}
+      />
+      <Field
+        control={form.control}
+        name="value"
+        label="Value"
+        render={({ field }) => (
+          <Input
+            {...field}
+            type="number"
+            onChange={(event) =>
+              field.onChange(event.currentTarget.valueAsNumber)
```

Comment:
This is unnecessary. `useForm` can handle numbers right away, without overriding `onChange`.